straxen/corrections_services.py

```diff
@@ -255,3 +297,36 @@ def cacheable_naming(*args, fmt='.npy', base='./resource_cache/'):
 class GainsNotFoundError(Exception):
     """Fatal error if a None value is returned by the corrections"""
     pass
+
+
+def get_cmt_local_versions(global_version):
+    cmt = CorrectionsManagementServices()
+    return cmt.get_local_versions(global_version)
+
+
+@strax.Context.add_method
+def apply_cmt_version(context, cmt_global_version, posrec_algo='mlp'):
+    """Sets all the relevant correction variables"""
+
+    local_versions = get_cmt_local_versions(cmt_global_version)
+
+    cmt_config = dict(gain_model=("to_pe_model", local_versions['tpc_gain_model'], True),
+                      gain_model_nv=("to_pe_model_nv", local_versions['nv_gain_model'], True),
+                      gain_model_mv=("to_pe_model_mv", local_versions['mv_gain_model'], True),
+                      s1_xyz_correction_map=(f's1_xyz_map_{posrec_algo}', local_versions[f's1_xyz_map_{posrec_algo}'], True),
+                      fdc_map=(f"fdc_map_{posrec_algo}", local_versions[f'fdc_map_{posrec_algo}'], True),
+                      s2_xy_correction_map=('s2_xy_map', local_versions['s2_xy_map'], True),
+                      elife_conf=("elife", local_versions['elife'], True),
+                      mlp_model=("mlp_model", local_versions['mlp_model'], True),
+                      gcn_model=("gcn_model", local_versions['gcn_model'], True),
+                      cnn_model=("cnn_model", local_versions['cnn_model'], True),
+                      electron_drift_velocity=("electron_drift_velocity", local_versions['electron_drift_velocity'], True),
+                      electron_drift_time_gate=("electron_drift_time_gate", local_versions['electron_drift_time_gate'], True),
+                      baseline_samples_nv=('baseline_samples_nv', local_versions['baseline_samples_nv'], True),
+                      hit_min_amplitude=('hit_thresholds_tpc', local_versions['hit_thresholds_tpc'], True),
+                      hit_min_amplitude_he=('hit_thresholds_he', local_versions['hit_thresholds_he'], True),
+                      hit_min_amplitude_nv=('hit_thresholds_nv', local_versions['hit_thresholds_nv'], True),
+                      hit_min_amplitude_mv=('hit_thresholds_mv', local_versions['hit_thresholds_mv'], True),
```

Comment:
@WenzDaniel I like the idea of this function. My only concern is that it relies on the options to have already been in CMT format before applying the function. Is this a safe assumption?

Reply:
I think so yes. I doubt that people will forget to change the setting after moving a correction to CMT. However, I think it is more likely that they will forget to update the hardcoded list above which will make a mess.

Reply:
Yeah I see your point, but this does not just depend on people updating default options right? While I don't see this happening much, one can imagine modifying a context in some way and then after that calling `apply_cmt_version`, where the changes that person made could affect how the version gets applied. It seems safer to me to just hardcode all the known corrections used for a given straxen version. I think the corrections experts/team leaders/ACs should be aware of this important function and make sure it gets updated if there is a new correction added (which hopefully doesn't happen much anyway). I would prefer this over assuming a certain convention that's not strictly enforced. 

 Or -- ideally -- sync the naming scheme between CMT and the strax plugins. But not sure this is doable now.

Reply:
>  would prefer this over assuming a certain convention that's not strictly enforced.
But it is is not it? 
If you specify e.g. `'baseline_samples_nv': 18` cmt will complain.

Reply:
> While I don't see this happening much, one can imagine modifying a context in some way and then after that calling apply_cmt_version, where the changes that person made could affect how the version gets applied. 

Could you give an example for such a modification? Why would we want to make a modification in xenonnt_online which gets overwritten in xenonnt? This does not make sense to me.

Reply:
I was under the impression you don't need to use CMT and that you can pass other options (like e.g. those old 1T maps stored in pax)?

Reply:
>Could you give an example for such a modification? Why would we want to make a modification in xenonnt_online which gets overwritten in xenonnt? This does not make sense to me.

I mean this is one of the main reasons we use strax to begin with right? The fact people can update configuration in their own context and reprocess quickly. Maybe someone is testing a new correction map and wants to switch between a new one not yet in CMT and an old one that is in CMT. 

So something like this (pseudocode)

```
st = xenonnt_online()
st.set_config('correction_map', 'some_new_correction_file')

# do something

# now want to apply global versions
st.apply_cmt_version('global_v1')
```
The apply_cmt_version in this case behaves differently depending on the state of the context passed to it. I think this is not ideal.

Reply:
I'm specifically concerned about someone calling `apply_cmt_version` outside of how the  `xenonnt` context gets initialized. It's an edge case so maybe not worth the time we are discussing here, but my view is that if we see a possible edge case issue that we should address it.

I also just don't think we should be assuming a convention in something as important as our corrections. I will propose an update to your function that doesn't make this assumption.

Reply:
edge cases are a "feature" of strax(en) not a "bug"... this is where I would say strax(en) is basically anarchy

Reply:
@ershockley , would making this function be not a context attribute an idea? Such that `st.apply_cmt_version` does not exist?

Reply:
> edge cases are a "feature" of strax(en) not a "bug"... this is where I would say strax(en) is basically anarchy

lol wtf no idea what to reply to this. Maybe "Write your own tool?" This is really a very unappropriated comment.

Reply:
Come on guys, no need for this. Let's end this conversation here. 

@jorana this could be an option as well. Let me try a few things and come back here later today (in a new discussion thread).

Reply:
@WenzDaniel I Did not mean to be disrespectful, my apologies if I gave the wrong impression... Let me rephrase it, since strax(en) allows user change any configs this potential leads to undesired challenges